lumicks/pylake/population/tests/conftest.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+import numpy as np
+import contextlib
+from lumicks.pylake.channel import Slice, Continuous
+from lumicks.pylake.population.detail.random import temp_seed
+
+
+def generate_trace(K, pi, A, mu, sigma, T, seed):
+    """Generate a time trace from HMM parameters.
+
+    Parameters
+    ----------
+    K : int
+        number of states
+    pi : initial state probability vector
+        array of shape (K, )
+    A : state transition probability matrix
+        array of shape (K, K)
+    mu : state means
+        array of shape (K, )
+    sigma : state standard deviations
+        array of shape (K, )
+    T : int
+        trace length
+    seed : int
+        random seed to use for generator
+    """
+    # make sure vectors are normalized
+    pi = pi / pi.sum()
+    A = A / A.sum(axis=1).reshape((-1, 1))
+
+    with temp_seed(seed):
+        # draw state path
+        sp = np.zeros(T, dtype=int)
+        sp[0] = np.random.choice(K, 1, replace=False, p=pi).squeeze()
+        for t in range(1, T):
+            sp[t] = np.random.choice(K, 1, replace=False, p=A[sp[t - 1]]).squeeze()
+
+        # draw observation vector
+        y = np.random.normal(loc=mu[sp], scale=sigma[sp])
```

Comment:
I'm honestly not sure how portable this seeded data generation is. E.g. if `numpy` change the way they generate random numbers in the future or if they use different generators on different platforms. Would the expected test values change in that case?

Reply:
True, but in this case I don't think it will matter really. I thought it was better to simulate more realistic data rather than hard-coding in a trace. This will especially be better when we get to dealing with HMMs. I think that as long as the simulated trace is long enough, the model fitting should converge to the input parameters, and the simulated statepath is also returned which is what is actually used for checking state assignments in the tests. I also thought setting the RNG seed was better than just leaving it completely uncontrolled.

I'm definitely open to suggestions if you think there is a better way to do this though.

Reply:
I think that the extremes are actually better here. If the data can be random, let it be truly random (no seeding). If the test can't tolerate true random data, generate the input once and store it. The middle ground is prone to issues that may show up in obscure circumstances.

Reply:
static file is the way to go I guess, data generation script is included in the test directory